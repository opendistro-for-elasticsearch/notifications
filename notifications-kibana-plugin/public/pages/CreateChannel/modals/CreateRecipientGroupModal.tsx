/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiButton,
  EuiButtonEmpty,
  EuiComboBoxOptionOption,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
} from '@elastic/eui';
import React, { useState } from 'react';
import { CreateRecipientGroupForm } from '../../Emails/forms/CreateRecipientGroupForm';

interface CreateSenderModalProps {
  props: any;
  onClose: () => void;
}

export function CreateRecipientGroupModal(props: any) {
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [selectedEmailOptions, setSelectedEmailOptions] = useState<
    Array<EuiComboBoxOptionOption<string>>
  >([]);
  const [emailOptions, setEmailOptions] = useState([
    {
      label: 'Titan',
    },
  ]);

  return (
    <EuiOverlayMask>
      <EuiModal onClose={props.onClose} style={{ width: 650 }}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>Create recipient group</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <CreateRecipientGroupForm
            name={name}
            setName={setName}
            description={description}
            setDescription={setDescription}
            selectedEmailOptions={selectedEmailOptions}
            setSelectedEmailOptions={setSelectedEmailOptions}
            emailOptions={emailOptions}
            setEmailOptions={setEmailOptions}
          />
        </EuiModalBody>

        <EuiModalFooter>
          <EuiButtonEmpty onClick={props.onClose} size="s">
            Cancel
          </EuiButtonEmpty>
          <EuiButton fill onClick={props.onClose} size="s">
            Create
          </EuiButton>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
}
