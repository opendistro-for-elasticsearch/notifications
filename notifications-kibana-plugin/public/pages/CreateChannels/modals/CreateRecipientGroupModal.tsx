import { ENCRYPTION_METHOD } from '../../../../models/interfaces';
import {
  EuiOverlayMask,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiModalBody,
  EuiTitle,
  EuiHorizontalRule,
  EuiModalFooter,
  EuiButton,
  EuiModal,
  EuiButtonEmpty,
  EuiFormRow,
  EuiFieldText,
  EuiFlexGroup,
  EuiFlexItem,
  EuiFieldNumber,
  EuiSpacer,
  EuiSuperSelect,
  EuiSuperSelectOption,
  EuiText,
  EuiTextArea,
  EuiComboBox,
  EuiComboBoxOptionOption,
} from '@elastic/eui';
import React, { useState } from 'react';

interface CreateSenderModalProps {
  props: any;
  onClose: () => void;
}

export function CreateRecipientGroupModal(props: any) {
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const [selectedEmailOptions, setSelectedEmailOptions] = useState<
    Array<EuiComboBoxOptionOption<string>>
  >([]);
  const [emailOptions, setEmailOptions] = useState([
    {
      label: 'Titan',
    },
  ]);

  const onCreateEmailOption = (
    searchValue: string,
    flattenedOptions: Array<EuiComboBoxOptionOption<string>> = []
  ) => {
    const normalizedSearchValue = searchValue.trim().toLowerCase();
    if (!normalizedSearchValue) return;

    const newOption = { label: searchValue };
    if (
      flattenedOptions.findIndex(
        (option) => option.label.trim().toLowerCase() === normalizedSearchValue
      ) === -1
    ) {
      setEmailOptions([...emailOptions, newOption]);
    }
    setSelectedEmailOptions([...selectedEmailOptions, newOption]);
  };

  return (
    <EuiOverlayMask>
      <EuiModal onClose={props.onClose} style={{ width: 650 }}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>Create recipient group</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiFormRow
            label="Name"
            style={{ maxWidth: '650px' }}
            helpText="The name must contain 2 to 50 characters. Valid characters are lowercase A-Z, a-z, 0-9, (_) underscore, (-) hyphen and unicode characters."
          >
            <EuiFieldText
              fullWidth
              placeholder="Enter channel name"
              value={name}
              onChange={(e) => setName(e.target.value)}
            />
          </EuiFormRow>

          <EuiSpacer size="m" />
          <EuiFormRow
            label={
              <span>
                Description - <i style={{ fontWeight: 'normal' }}>optional</i>
              </span>
            }
            style={{ maxWidth: '650px' }}
          >
            <>
              <EuiText size="xs" color="subdued">
                Describe the purpose of the channel.
              </EuiText>
              <EuiTextArea
                fullWidth
                placeholder="Description"
                style={{ height: '2.8rem' }}
                value={description}
                onChange={(e) => setDescription(e.target.value)}
              />
            </>
          </EuiFormRow>

          <EuiSpacer size="m" />
          <EuiFormRow label="Emails" style={{ maxWidth: '650px' }}>
            <EuiComboBox
              placeholder="Email addresses"
              fullWidth
              options={emailOptions}
              selectedOptions={selectedEmailOptions}
              onChange={setSelectedEmailOptions}
              onCreateOption={onCreateEmailOption}
              isClearable={true}
            />
          </EuiFormRow>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiButtonEmpty onClick={props.onClose} size="s">
            Cancel
          </EuiButtonEmpty>
          <EuiButton fill onClick={props.onClose} size="s">
            Create
          </EuiButton>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
}
