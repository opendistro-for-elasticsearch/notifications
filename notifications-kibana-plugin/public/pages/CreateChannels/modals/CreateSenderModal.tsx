import { ENCRYPTION_METHOD } from '../../../../models/interfaces';
import {
  EuiOverlayMask,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiModalBody,
  EuiTitle,
  EuiHorizontalRule,
  EuiModalFooter,
  EuiButton,
  EuiModal,
  EuiButtonEmpty,
  EuiFormRow,
  EuiFieldText,
  EuiFlexGroup,
  EuiFlexItem,
  EuiFieldNumber,
  EuiSpacer,
  EuiSuperSelect,
  EuiSuperSelectOption,
} from '@elastic/eui';
import React, { useState } from 'react';

interface CreateSenderModalProps {
  props: any;
  onClose: () => void;
}

export function CreateSenderModal(props: any) {
  const [senderName, setSenderName] = useState('');
  const [email, setEmail] = useState('');
  const [host, setHost] = useState('');
  const [port, setPort] = useState('465');
  const [encryption, setEncryption] = useState<ENCRYPTION_METHOD>('SSL');
  const encryptionOptions: Array<EuiSuperSelectOption<ENCRYPTION_METHOD>> = [
    {
      value: 'SSL',
      inputDisplay: 'SSL',
    },
    {
      value: 'TSL',
      inputDisplay: 'TSL',
    },
  ];

  return (
    <EuiOverlayMask>
      <EuiModal onClose={props.onClose} style={{ width: 650 }}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>Create sender</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiFormRow
            label="Sender name"
            style={{ maxWidth: '650px' }}
            helpText="Use a unique and descriptive name that's easy to search. The sender name must contain from m to n characters. Valid characters are lowercase a-z, 0-9, and - (hyphen)."
          >
            <EuiFieldText
              fullWidth
              placeholder=""
              value={senderName}
              onChange={(e) => setSenderName(e.target.value)}
            />
          </EuiFormRow>

          <EuiSpacer size="m" />
          <EuiFlexGroup gutterSize="s">
            <EuiFlexItem grow={4}>
              <EuiFormRow label="Email address">
                <EuiFieldText
                  placeholder=""
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                />
              </EuiFormRow>
            </EuiFlexItem>
            <EuiFlexItem grow={4}>
              <EuiFormRow label="Host">
                <EuiFieldText
                  placeholder=""
                  value={host}
                  onChange={(e) => setHost(e.target.value)}
                />
              </EuiFormRow>
            </EuiFlexItem>
            <EuiFlexItem grow={2}>
              <EuiFormRow label="Port">
                <EuiFieldNumber
                  placeholder=""
                  value={port}
                  onChange={(e) => setPort(e.target.value)}
                />
              </EuiFormRow>
            </EuiFlexItem>
          </EuiFlexGroup>

          <EuiSpacer size="m" />
          <EuiFormRow
            label="Encryption method"
            style={{ maxWidth: '700px' }}
            helpText={
              <div>
                SSL or TSL is recommended for security. SSL and TSL requires
                validation by adding the following two fields to Elasticsearch
                keystore:
                <br />
                opendistro.alerting.destination.mail.adminTest.username:
                [username]
                <br />
                opendistro.alerting.destination.mail.adminTest.password:
                [password]
              </div>
            }
          >
            <EuiSuperSelect
              fullWidth
              options={encryptionOptions}
              valueOfSelected={encryption}
              onChange={setEncryption}
            />
          </EuiFormRow>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiButtonEmpty onClick={props.onClose} size="s">
            Cancel
          </EuiButtonEmpty>
          <EuiButton fill onClick={props.onClose} size="s">
            Create
          </EuiButton>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
}
