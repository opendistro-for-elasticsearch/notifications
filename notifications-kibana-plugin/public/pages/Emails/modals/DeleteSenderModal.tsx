/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import {
  EuiButton,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiTitle,
  EuiHorizontalRule,
  EuiText,
  EuiFlexItem,
  EuiFlexGroup,
  EuiButtonEmpty,
} from '@elastic/eui';
import { SenderItemType } from '../../../../models/interfaces';

interface DeleteSenderModalProps {
  senders: SenderItemType[];
  onClose: () => void;
}

export const DeleteSenderModal = ({
  senders,
  onClose,
}: DeleteSenderModalProps) => {
  if (!senders.length) return null;

  const plural = senders.length >= 2;
  const name = plural ? `${senders.length} senders` : senders[0].name;
  const message = `Delete ${name} permanently? Any channels using ${
    plural ? 'these' : 'this'
  } email sender${plural ? 's' : ''} will not be able to send notifications.`;

  return (
    <EuiOverlayMask>
      <EuiModal onClose={onClose} maxWidth={500}>
        <EuiModalHeader>
          <EuiModalHeaderTitle>{`Delete ${name}?`}</EuiModalHeaderTitle>
        </EuiModalHeader>
        <EuiModalBody>
          <EuiText>{message}</EuiText>
        </EuiModalBody>
        <EuiModalFooter>
          <EuiFlexGroup justifyContent="flexEnd">
            <EuiFlexItem grow={false}>
              <EuiButtonEmpty onClick={onClose}>Cancel</EuiButtonEmpty>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <EuiButton fill color="danger" onClick={onClose}>
                Delete
              </EuiButton>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
};
