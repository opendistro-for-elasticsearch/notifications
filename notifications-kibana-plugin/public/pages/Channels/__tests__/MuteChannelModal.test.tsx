/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { MuteChannelModal } from '../components/modals/MuteChannelModal';
import { notificationServiceMock } from '../../../../test/mocks/serviceMock';

describe('<MuteChannelModal /> spec', () => {
  it('returns if no channels', () => {
    const { container } = render(
      <MuteChannelModal
        channels={[]}
        onClose={() => {}}
        services={notificationServiceMock}
        mute={true}
      />
    );
    expect(container.firstChild).toBeNull();
  });

  it('renders the component', () => {
    const channels = [jest.fn() as any];
    const { container } = render(
      <MuteChannelModal
        channels={channels}
        onClose={() => {}}
        services={notificationServiceMock}
        mute={true}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
