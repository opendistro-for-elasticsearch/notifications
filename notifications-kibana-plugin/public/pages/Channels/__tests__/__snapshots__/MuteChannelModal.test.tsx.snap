// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuteChannelModal /> spec renders the component 1`] = `null`;
